### PR TITLE
Corrections and improvements to Eclipse compatible restart file output 

### PR DESCRIPTION
**Pull request by jalvestad**
Some logical tests in the code has been corrected / improved - testing ongoing

### CODE REVIEW

**Comment by jalvestad** [comment]
This PR is now ready from me for final review and merging into master.

**Comment by bska** [comment]
jenkins build this please

File: src/opm/output/eclipse/AggregateGroupData.cpp [hunk 1]
```diff
@@ -716,7 +716,7 @@ void staticContrib(const Opm::Schedule&     sched,
                         iGrp[nwgmax + 17] = 0;
                     }
                     else if (higher_lev_winj_ctrl > 0 || higher_lev_winj_cmode > 0) {
-                        if ((winj_cmode != Opm::Group::InjectionCMode::FLD) || (winj_cmode!= Opm::Group::InjectionCMode::NONE)) {
+                        if (!((winj_cmode == Opm::Group::InjectionCMode::FLD) || (winj_cmode == Opm::Group::InjectionCMode::NONE))) {
```
Comment (bska) [inline src/opm/output/eclipse/AggregateGroupData.cpp:719]: This change **appears** to influence `IGRP[nwgmax + 17]` for the `CENTRAL` group of [MOD4_GRP](https://ci.opm-project.org/job/opm-common-PR-builder/2304/testReport/junit/(root)/mpi/compareECLFiles_flow_MOD4_GRP/) and [MOD4_UDQ_ACTIONX]() on report steps 38 and 51.  The value is reset from `3` to `1` in both test cases.

Would you be able to confirm/verify that this behaviour is expected?

Comment (jalvestad) [reply src/opm/output/eclipse/AggregateGroupData.cpp:719]: Yes - this change is OK/expected and gives the value it should have in this case!

Comment (bska) [reply src/opm/output/eclipse/AggregateGroupData.cpp:719]: > Yes - this change is OK/expected and gives the value it should have in this case!

Excellent.  Thanks a lot for confirming the expected behaviour.

**Review by bska — APPROVED** [review]
Thanks a lot for these updates.  I'll update the reference solutions once I'm sure that doing so won't conflict with other potential updates.

**Comment by bska** [comment]
> I'll update the reference solutions once I'm sure that doing so won't conflict with other potential updates.

While we investigate the possible regression in OPM/opm-simulators#2701 I am going to update the reference solutions of MOD4_GRP and MOD4_UDQ_ACTIONX to prepare for merging this PR.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=351 please

**Comment by bska** [comment]
The reference data update has been merged.  I'm merging this now to bring the baseline up-to-date with the reference solutions.  Thanks a lot for the patch!

**Pull request merged**